### ISSUE TITLE
Update fix hash and directory

### ISSUE THREAD

**Issue by RussTreadon-NOAA** [NOAA-EMC/GSI#580]
`develop` pulls fix files from two locations:
- ASCII GSI fix files are accessed via submodule `fix` which points at a unique hash in [GSI-Fix](https://github.com/NOAA-EMC/GSI-fix)
- Binary GSI fix files are pulled into `fix` during the build process.  Binary GSI fix files are accessed from the path, `GSI_BINARY_SOURCE_DIR` found in `modulefiles/gsi_${machine}.lua`.   If `GSI_BINARY_SOURCE_DIR` is not defined, the binary fix files are downloaded from `https://ftp.emc.ncep.noaa.gov/static_files/public/gsi`.

With the above background this issue is opened to address recent GSI fix file changes.

1. GSI-Fix [PR #8](https://github.com/NOAA-EMC/GSI-fix/pull/8) updated `global_convinfo.txt` to be consistent with the operational GFS v16.3.7.   With merger of PR [#8](https://github.com/NOAA-EMC/GSI-fix/pull/8) into GSI-fix `develop`, the fix hash at which GSI `develop` points at needs to be updated.
2. EIB create a new `GSI_BIN_FIX_DIR`.   The path for `GSI_BIN_FIX_DIR` needs to be updated in GSI modulefiles.  The need to do this was raised by @KateFriedman-NOAA in [this comment](https://github.com/NOAA-EMC/global-workflow/issues/1550#issuecomment-1550337199) from g-w issue [#1550](https://github.com/NOAA-EMC/global-workflow/issues/1550)
3. A new static_files/public/gsi tarball needs to be created consistent with the new `GSI_BIN_FIX_DIR`.  An issue noting this will be opened in g-w.

**Comment by RussTreadon-NOAA** [comment]
I downloaded `gsi-fix-b90f7b.tgz` from [ftp.emc.ncep.noaa.gov/static_files/public/gsi](https://ftp.emc.ncep.noaa.gov/static_files/public/gsi).   After a gunzip and tar -xvf, comparison of the contents of the tarball show now differences with like files in `/scratch1/NCEPDEV/global/glopara/fix/gsi/20230112/`.    

A review of this issue shows that only ASCII GSI fix files were updated in 20230112.   This is consistent with comparison of the tarball contents with 20230112.   As such, we can keep using the binary GSI tarball `gsi-fix-b90f7b.tgz` posted at  [ftp.emc.ncep.noaa.gov/static_files/public/gsi](https://ftp.emc.ncep.noaa.gov/static_files/public/gsi).  As such, `3` above does not require any action at present.  No g-w issue will be opened.

It's challenging to keep everything aligned when hosting GSI fix files in various locations with a variety of approaches as [noted here](https://github.com/NOAA-EMC/global-workflow/issues/1550#issuecomment-1539003531).

**Comment by RussTreadon-NOAA** [comment]
Work for this issue will be done in [`RussTreadon-NOAA:feature/gsifix`](https://github.com/RussTreadon-NOAA/GSI/tree/feature/gsifix)

**Comment by RussTreadon-NOAA** [comment]
Hera ctests
```
Hera(hfe01):/scratch1/NCEPDEV/da/Russ.Treadon/git/gsi/gsifix/build$ ctest -j 9
Test project /scratch1/NCEPDEV/da/Russ.Treadon/git/gsi/gsifix/build
    Start 1: global_3dvar
    Start 2: global_4dvar
    Start 3: global_4denvar
    Start 4: hwrf_nmm_d2
    Start 5: hwrf_nmm_d3
    Start 6: rtma
    Start 7: rrfs_3denvar_glbens
    Start 8: netcdf_fv3_regional
    Start 9: global_enkf
1/9 Test #7: rrfs_3denvar_glbens ..............***Failed  1690.65 sec
2/9 Test #9: global_enkf ......................   Passed  1969.33 sec
3/9 Test #3: global_4denvar ...................   Passed  3476.14 sec
4/9 Test #8: netcdf_fv3_regional ..............***Failed  3489.02 sec
5/9 Test #4: hwrf_nmm_d2 ......................   Passed  3491.10 sec
6/9 Test #5: hwrf_nmm_d3 ......................   Passed  3502.97 sec
7/9 Test #6: rtma .............................   Passed  4034.20 sec
8/9 Test #1: global_3dvar .....................   Passed  5036.45 sec
9/9 Test #2: global_4dvar .....................   Passed  5227.11 sec

78% tests passed, 2 tests failed out of 9

Total Test time (real) = 5227.12 sec

The following tests FAILED:
          7 - rrfs_3denvar_glbens (Failed)
          8 - netcdf_fv3_regional (Failed)
Errors while running CTest
Output from these tests are in: /scratch1/NCEPDEV/da/Russ.Treadon/git/gsi/gsifix/build/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
```
The `rrfs_3denvar_glbens` failure is due to 
```
The case has Failed the scalability test.
The slope for the update (30.688913 seconds per node) is less than that for the control (47.165411 seconds per node).
```
A check of the the wall times
```
rrfs_3denvar_glbens_hiproc_contrl/stdout:The total amount of wall time                        = 80.837799
rrfs_3denvar_glbens_hiproc_updat/stdout:The total amount of wall time                        = 101.318688
rrfs_3denvar_glbens_loproc_contrl/stdout:The total amount of wall time                        = 128.003210
rrfs_3denvar_glbens_loproc_updat/stdout:The total amount of wall time                        = 125.869819
```
shows the hiproc_updat time to be noticeably larger than the control.   A rerun of the `rrfs_3denvar_glbens` yielded a pass

```
Hera(hfe01):/scratch1/NCEPDEV/da/Russ.Treadon/git/gsi/gsifix/build$ ctest -R rrfs_3denvar_glbens
Test project /scratch1/NCEPDEV/da/Russ.Treadon/git/gsi/gsifix/build
    Start 7: rrfs_3denvar_glbens
1/1 Test #7: rrfs_3denvar_glbens ..............   Passed  550.38 sec

100% tests passed, 0 tests failed out of 1

Total Test time (real) = 550.39 sec
```


The `netcdf_fv3_regional` failure is a known problem.  The loproc and hiproc configurations yield different analysis results.  PR #570 contains code changes which fix this problem.

**Comment by RussTreadon-NOAA** [comment]
**WCOSS2 (Cactus) ctests**
```
russ.treadon@clogin09:/lfs/h2/emc/da/noscrub/russ.treadon/git/gsi/gsifix/build> ctest -j 9
Test project /lfs/h2/emc/da/noscrub/russ.treadon/git/gsi/gsifix/build
    Start 1: global_3dvar
    Start 6: rtma
    Start 8: netcdf_fv3_regional
    Start 2: global_4dvar
    Start 3: global_4denvar
    Start 7: rrfs_3denvar_glbens
    Start 4: hwrf_nmm_d2
    Start 9: global_enkf
    Start 5: hwrf_nmm_d3
1/9 Test #8: netcdf_fv3_regional ..............***Failed  737.81 sec
2/9 Test #9: global_enkf ......................   Passed  868.02 sec
3/9 Test #7: rrfs_3denvar_glbens ..............   Passed  991.11 sec
4/9 Test #5: hwrf_nmm_d3 ......................   Passed  1652.52 sec
5/9 Test #4: hwrf_nmm_d2 ......................***Failed  1820.44 sec
6/9 Test #1: global_3dvar .....................   Passed  1827.79 sec
7/9 Test #2: global_4dvar .....................   Passed  2056.14 sec
8/9 Test #3: global_4denvar ...................   Passed  2120.43 sec
9/9 Test #6: rtma .............................   Passed  2129.29 sec

78% tests passed, 2 tests failed out of 9

Total Test time (real) = 2129.46 sec

The following tests FAILED:
          4 - hwrf_nmm_d2 (Failed)
          8 - netcdf_fv3_regional (Failed)
Errors while running CTest
russ.treadon@clogin09:/lfs/h2/emc/da/noscrub/russ.treadon/git/gsi/gsifix/build> 
```

The `hwrf_nmm_d2` failure is due to the scalability test.   Wall times for this test were
```
hwrf_nmm_d2_hiproc_contrl/stdout:The total amount of wall time                        = 100.417084
hwrf_nmm_d2_hiproc_updat/stdout:The total amount of wall time                        = 116.694299
hwrf_nmm_d2_loproc_contrl/stdout:The total amount of wall time                        = 131.751604
hwrf_nmm_d2_loproc_updat/stdout:The total amount of wall time                        = 136.535021
```
The loproc minus hiproc wall time difference for the update is less than the control.   This may be due to system issues when the various jobs ran.   This is not a fatal fail.   A rerun of the `hwrf_nmm_d2` test returned a pass

```
russ.treadon@clogin09:/lfs/h2/emc/da/noscrub/russ.treadon/git/gsi/gsifix/build> ctest -R hwrf_nmm_d2
Test project /lfs/h2/emc/da/noscrub/russ.treadon/git/gsi/gsifix/build
    Start 4: hwrf_nmm_d2
1/1 Test #4: hwrf_nmm_d2 ......................   Passed  1288.88 sec

100% tests passed, 0 tests failed out of 1

Total Test time (real) = 1288.88 sec
```


The `netcdf_fv3_regional` failure is a known problem.  Source code updates in PR #570 fix this ctest.

**Comment by RussTreadon-NOAA** [comment]
**Orion ctests**
```
Orion-login-3:/work2/noaa/da/rtreadon/git/gsi/gsifix/build$ ctest -j 9
Test project /work2/noaa/da/rtreadon/git/gsi/gsifix/build
    Start 1: global_3dvar
    Start 2: global_4dvar
    Start 3: global_4denvar
    Start 4: hwrf_nmm_d2
    Start 5: hwrf_nmm_d3
    Start 6: rtma
    Start 7: rrfs_3denvar_glbens
    Start 8: netcdf_fv3_regional
    Start 9: global_enkf
1/9 Test #9: global_enkf ......................   Passed  491.16 sec
2/9 Test #7: rrfs_3denvar_glbens ..............   Passed  546.48 sec
3/9 Test #4: hwrf_nmm_d2 ......................   Passed  608.23 sec
4/9 Test #8: netcdf_fv3_regional ..............***Failed  663.28 sec
5/9 Test #5: hwrf_nmm_d3 ......................   Passed  856.51 sec
6/9 Test #6: rtma .............................   Passed  1211.50 sec
7/9 Test #3: global_4denvar ...................***Failed  1624.14 sec
8/9 Test #2: global_4dvar .....................   Passed  1741.91 sec
9/9 Test #1: global_3dvar .....................   Passed  1923.68 sec

78% tests passed, 2 tests failed out of 9

Total Test time (real) = 1923.69 sec

The following tests FAILED:
          3 - global_4denvar (Failed)
          8 - netcdf_fv3_regional (Failed)
Errors while running CTest
Output from these tests are in: /work2/noaa/da/rtreadon/git/gsi/gsifix/build/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
```
The `global_4denvar` failure is due to
```
The runtime for global_4denvar_loproc_updat is 407.925066 seconds.  This has exceeded maximum allowable threshold time of 405.003122 seconds,
resulting in Failure time-thresh of the regression test.
```
A rerun of `global_4denvar` resulted in a pass
```
Orion-login-3:/work2/noaa/da/rtreadon/git/gsi/gsifix/build$ ctest -R global_4denvar
Test project /work2/noaa/da/rtreadon/git/gsi/gsifix/build
    Start 3: global_4denvar
1/1 Test #3: global_4denvar ...................   Passed  1563.86 sec

100% tests passed, 0 tests failed out of 1

Total Test time (real) = 1563.87 sec
```
The `netcdf_fv3_regional` failure is expected and will be fixed by #570.
